src/layers/stateless.jl

```diff
@@ -1,3 +1,240 @@
+"""
+    mae(ŷ, y; agg=mean)
+
+Return the loss corresponding to mean absolute error: 
+
+    agg(abs.(ŷ .- y))
+"""
+mae(ŷ, y; agg=mean) = agg(abs.(ŷ .- y))
+
+"""
+    mse(ŷ, y; agg=mean)
+
+Return the loss corresponding to mean square error: 
+    
+    agg((ŷ .- y).^2)
+"""
+mse(ŷ, y; agg=mean) = agg((ŷ .- y).^2)
+
+"""
+    msle(ŷ, y; agg=mean, ϵ=eps(ŷ))
+
+The loss corresponding to mean squared logarithmic errors, calculated as
+
+    agg((log.(ŷ .+ ϵ) .- log.(y .+ ϵ)).^2)
+
+The `ϵ` term provides numerical stability.
+Penalizes an under-estimation more than an over-estimatation.
+"""
+msle(ŷ, y; agg=mean, ϵ=epseltype(ŷ)) = agg((log.(ŷ .+ ϵ) .- log.(y .+ ϵ)).^2)
```

Comment:
This is more accurate:

```suggestion
msle(ŷ, y; agg=mean, ϵ=epseltype(ŷ)) = agg((log.((ŷ .+ ϵ) ./ (y .+ ϵ)).^2)
```

For example,

```julia
julia> log(1e9) - log(1e9 + 1)
-1.000000082740371e-9
julia> log(1e9/(1 + 1e9))
-9.999999722180686e-10
```

The first form has an error of `~1e-16` while the later error is `1e-17`. (Exact answer is 9.99999999500000000333333333083e-10)

Reply:
done!